api/src/main/java/org/datacleaner/api/WSStatelessComponent.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.api;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Components with the {@link org.datacleaner.api.WSStatelessComponent} annotation are provided
+ * in a web service for components.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.TYPE })
+public @interface WSStatelessComponent {
```

Comment:
Well, we think that supposing all components are stateless and provided in REST API implicitly is a potential problem. Firstly, some transformers have their state (for example ID Generator). Secondly, not all components are required to be available in the REST API. So if someone marks the component stateless by the annotation it is his responsibility that it is not stateless and accessible. ("Do what is allowed" politics instead of "do not do what is not allowed". )